api/src/decorators/json-api-parameters.decorator.ts

```diff
@@ -157,3 +157,19 @@ export const JSONAPISingleEntityQueryParams = (fetchConfiguration?: {
     fieldsQueryParam,
     omitFieldsQueryParam,
   );
+
+export const JSONAPISourcingMaterialsQueryParams = (fetchConfiguration?: {
```

Comment:
This seems to be a subset of  `JSONAPIQueryParams`, removing a few decorators that seem to do nothing by default, so it's not clear why we need it. Also, even if we do, the name is domain-specific, so if we do decide to keep it, it needs to be renamed

Reply:
Here I tried to exclude the query params which may have unexpected behavior when endpoint returns materials with 'materialsData=true' (such as omit, sort, field). Those flags come from base-service and work with the columns of the original entity table (sourcing_locations), but not with the joined ones (for example, we cannot filter by material name, because 'material' comes as an array of joined relations, it is not a column of 'sourcing_locations' table).
However, since JSONAPIQueryParams is still correct for this endpoint without 'materialsData=true', I would probably leave it and try to document the above mentioned details..

Reply:
kudos for testing out those scenarios ;)

ok, but taking your example, and since this endpoint would return SLs, not Materials, the "omit" would only apply to the SL fields, not the Material fields, right?

If you still think the slimmed down version of the decorator is the best option, the go for it, but use a different name for it, so it describes the function of the decorator, not where it's used.